extension/api.js

```diff
@@ -43,25 +56,51 @@
 	})();
 
 	window.gitHubNotifCount = function (callback) {
-		var tmp = document.createElement('div');
+		var token = GitHubNotify.settings.get('oauthToken');
+		var opts = {
+			Authorization: 'token ' + token
+		};
+		var participating = GitHubNotify.settings.get('useParticipatingCount')
+			? '?participating=true'
+			: '';
+		var url = GitHubNotify.settings.get('rootUrl');
 
-		xhr('GET', GitHubNotify.settings.get('notificationUrl'), function (data, status) {
+		if (!token) {
+			callback(new Error('missing token'));
+			return;
+		}
+		if (/(^(https:\/\/)?(api\.)?github\.com)/.test(url)) {
+			url = 'https://api.github.com/notifications';
+		} else {
+			url += 'api/v3/notifications';
+		}
+		url += participating;
+
+		xhr('GET', url, opts, function (data, status, response) {
+			var interval = Number(response.getResponseHeader('X-Poll-Interval'));
 			if (status >= 400) {
```

Comment:
Note that a bad token would probably result in a 401, and it is technically not a server error. Maybe a `>= 500` would be more acceptable here, and a separate check for `>= 400` or even one for `== 401`.